lib/ruby_lsp_rails/railtie.rb

```diff
@@ -1,38 +0,0 @@
-# typed: strict
-# frozen_string_literal: true
-
-require "rails"
-require "ruby_lsp_rails/rack_app"
-
-module RubyLsp
-  module Rails
-    class Railtie < ::Rails::Railtie
-      config.ruby_lsp_rails = ActiveSupport::OrderedOptions.new
```

Comment:
Let's maybe keep the railtie so we can print deprecation warning for apps assigning `config.ruby_lsp_rails.server`?

I think upgrading `ruby-lsp-rails` and having the app blown up is not a good impression we want users to have.

Reply:
I was thinking that originally, but after discussing with @vinistock, probably very few apps are likely to be using this.

For Core, we can remove the setting when updating the gem.

We're also tagging this PR as a 'breaking change' for awareness.

Reply:
> We're also tagging this PR as a 'breaking change' for awareness.

Yes, this and the version change will technically justify it, but they won't show up in the users' bump PR I think.

Given that:
1. This addon was [massively promoted](https://twitter.com/excid3/status/1753245795615125784) by Chris Oliver recently
2. Most Rails apps are private and won't show up in the GH search result

I don't want to underestimate the negative effect this may cause to our users, especially when it's actually not too costly to avoid.

Reply:
I highly doubt a lot of people are turning off the server, but if you feel strongly about this we can add a deprecation warning and then remove it later.

Reply:
Done (and verified against Code DB).

Reply:
Thanks 👍 

I want to be super careful about this because I did already find 2 (though inactive) repos having this set just doing a [quick search](https://github.com/search?q=config.ruby_lsp_rails.server&type=code).